app/frontend/src/lib/ui/components/filterGroup.js

```diff
@@ -27,17 +30,41 @@ export const init = (groupContainerSelector, removeButtonSelector, clearButtonSe
       document.getElementsByClassName('moj-filter-layout__content')[0].classList.toggle('govuk-grid-column-three-quarters');
 
       if (document.getElementsByClassName('moj-filter-sidebar')[0].classList.contains('moj-filter-sidebar__hidden')) {
-        toggleButton.innerHTML = 'Show filters';
+        toggleButton.innerText = 'Show filters';
       } else {
-        toggleButton.innerHTML = 'Hide filters';
+        toggleButton.innerText = 'Hide filters';
       }
     });
   }
+
+  if (document.getElementById(mobileFiltersButtonSelector)) {
+    document.getElementById(mobileFiltersButtonSelector).addEventListener('click', () => {
+      Array.from(document.getElementsByClassName('filters--hide-mobile')).map((element) => element.classList.toggle('filters--show-mobile'));
+    });
+  }
 };
 
-export const closeAllSectionsHandler = (e) => {
+export const displayOpenOrCloseText = (targetElement, expandedElements) => {
+  if (targetElement.innerText === CLOSE_ALL_TEXT && expandedElements === 0) {
+    targetElement.innerText = OPEN_ALL_TEXT;
+  } else if (targetElement.innerText === OPEN_ALL_TEXT && expandedElements > 0) {
+    targetElement.innerText = CLOSE_ALL_TEXT;
+  }
+};
+
+export const openOrCloseAllSectionsHandler = (e) => {
   e.preventDefault();
```

Comment:
i dont think default needs to be prevented as all of these are buttons now (without type="submit" attribute and not a tags)